hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegistryEndpointsRefresher.java

```diff
@@ -60,9 +63,20 @@ public synchronized void stop() {
     notifyAll();
   }
 
+  private long getRefreshIntervalMs(boolean firstRefresh) {
+    if (refreshNow) {
+      return minTimeBetweenRefreshesMs;
+    }
+    if (firstRefresh) {
```

Comment:
I think this works ok for a large cluster once we populate the stubs of all the bootstrap nodes. But for a smaller cluster (<=10), populateStubs() returns everything. So its likely that even after the first refresh, all the subsequent refreshes hit the same bunch of nodes at the same time. 

It looks to me like the right way to fix this is to add jitter to the refreshTime? jitter is a random sleep between (0, n) so that we spread out the calls, this is for _all_ the refreshes and not just the first refresh, WDYT?

Reply:
I think jitter is used to solve another problem? It is always good to add a jitter to periodically requests to reduce the possibility to request at the same time.

For the purpose of this issue, let me just paste the description here first to see it can help you to understand better

> Introduce a initial refresh interval for RpcConnectionRegistry  so we can get the new list soon once we connect to the cluster.
As end users could configure any nodes in a cluster as the initial bootstrap nodes, it is possible that different end users will configure the same machine which makes the machine over load. So we should have a shorter delay for the initial refresh, to let users quickly switch to the bootstrap nodes we want them to connect to.

Free feel to ask if you have other concerns.

Reply:
Oh, sorry interpreted it as something else totally. I thought you were talking about what if all requests hit the servers at the same time and suggested jitter, my bad!